src/apps/api/router.py

```
@@ -70,29 +118,191 @@
+
+            # Clear existing index
+            yield f"data: {json.dumps({'type': 'status', 'message': 'Clearing existing index...', 'progress': 10})}\n\n"
+
+            import shutil
+            from pathlib import Path
+
+            db_path = coordinator.vector_store.persist_directory
+            if Path(db_path).exists():
+                try:
+                    shutil.rmtree(db_path)
+                except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)